### DISCUSSION TITLE
404 not found my website is not publishing

### DISCUSSION THREAD

**Discussion by ssc-gov** [community/community#22106] [Other Feature Feedback, Questions, & Ideas]
<p>i have a repository  <a href="https://captiongram.github.io/caption.github.io/" rel="nofollow noopener">https://captiongram.github.io/caption.github.io/</a> but when  ever i open it ,they show 404 error not found can you please help me to publish my website its not publishing i waited for more than an hour.</p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/9/94d134eb443a4ba1b0d8a7adc9ce9bdf5d2a7107.jpeg" data-download-href="/uploads/short-url/leuT76HsW5m7395RdxVPXBg5GbZ.jpeg?dl=1" title="Untitled1" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/65845248/181068892-9a700a74-7e3f-4612-b4ad-93e4f074da69.jpeg" alt="Untitled1" data-base62-sha1="leuT76HsW5m7395RdxVPXBg5GbZ" width="690" height="386" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/9/94d134eb443a4ba1b0d8a7adc9ce9bdf5d2a7107_2_690x386.jpeg, https://aws1.discourse-cdn.com/github/optimized/2X/9/94d134eb443a4ba1b0d8a7adc9ce9bdf5d2a7107_2_1035x579.jpeg 1.5x, https://aws1.discourse-cdn.com/github/original/2X/9/94d134eb443a4ba1b0d8a7adc9ce9bdf5d2a7107.jpeg 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/9/94d134eb443a4ba1b0d8a7adc9ce9bdf5d2a7107_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">Untitled1</span><span class="informations">1366×765 83.2 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>

**Comment by ssc-gov** [comment]
<p>this one is my url showing in repository settings<br>
</p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/e/eb63b283253cb4faff02b4029dd361420d6d67ac.png" data-download-href="/uploads/short-url/xAlNYvOZnrxkk0BGOiIv1C4BnOs.png?dl=1" title="Untitled" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/65845248/181068893-cf366f66-be5a-4a37-9d26-5029a6c4977d.png" alt="Untitled" data-base62-sha1="xAlNYvOZnrxkk0BGOiIv1C4BnOs" width="690" height="387" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/e/eb63b283253cb4faff02b4029dd361420d6d67ac_2_690x387.png, https://aws1.discourse-cdn.com/github/optimized/2X/e/eb63b283253cb4faff02b4029dd361420d6d67ac_2_1035x580.png 1.5x, https://aws1.discourse-cdn.com/github/original/2X/e/eb63b283253cb4faff02b4029dd361420d6d67ac.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/e/eb63b283253cb4faff02b4029dd361420d6d67ac_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">Untitled</span><span class="informations">1366×768 73.8 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>

**Comment by Andre601** [comment]
<p>If you want a user page (One that shows under <code>:username.github.io</code>) will you need to take some specific steps:</p>
<ol>
<li>The repo name has to match <code>:username.github.io</code> (<code>:username</code> would be your username, not displayed name)</li>
<li>The content has to be on the <code>master</code> branch. User pages don’t accept the <code>gh-pages</code> branch or the <code>docs</code> folder as alternatives… Idk why that is.<br>
The page also only uses valid files which afaiak are markdown, html, css and javascript files and images.</li>
</ol>
<p>It also may take some time for GitHub to actually “build” your page and make it ready, so you should try it after 10 minutes or so.</p>
<p>If you however want to have a separate page (which would appear under <code>:username.github.io/:repository</code>) will you need to make sure, that you provide valid files in the right directions.<br>
Those kind of pages allow you to use the <code>gh-pages</code> branch (often recommendet), the <code>docs</code> folder on the <code>master</code> branch or just the <code>master</code> branch itself (which is your case right now).</p>
<p>Same rules in terms of allowed files and build time apply here too.</p>
<p>I hope I was able to answer some questions.</p>

**Comment by indigofficial** [comment, marked as answer]
<p><a class="mention" href="/captiongram">@captiongram</a> you have to enter the direct link, like what <a class="mention" href="/Andre601">@Andre601</a> said. You have to enter the folder’s name, and then the your file’s name.</p>
<p>This is the working link:<br>
<a href="https://captiongram.github.io/captiongram.github.io/captiongram/home" class="onebox" target="_blank" rel="nofollow noopener">https://captiongram.github.io/captiongram.github.io/captiongram/home</a></p>
<p>Your username is <strong>captiongram</strong>, your respository is <strong><a href="http://captiongram.github.io" rel="nofollow noopener">captiongram.github.io</a></strong> but you need to also enter your folder name in the link, as GitHub does not do this for you.</p>
<p>It’s also best if you put images and HTML files in separate folders. :slight_smile:</p>

**Comment by ssc-gov** [comment]
<p>thnkxx its working now i can see my website online</p>

**Comment by indigofficial** [comment]
<p>Your welcome, <a class="mention" href="/captiongram">@captiongram</a>! I’m glad it helped :grin:</p>

**Comment by ssc-gov** [comment]
<p>again i’m in trouble .In my webpage there is a button “services” in that i have put 3 options “whatsapp”,“Facebook”,“Instagram” they are not working can u please help!!<br>
<a href="https://captiongram.github.io/captiongram.github.io/captiongram/services.html" class="onebox" target="_blank" rel="nofollow noopener">https://captiongram.github.io/captiongram.github.io/captiongram/services.html</a></p>
<p>next  links i’am not able to paste due to new user guidelines<br>
<a href="/uploads/short-url/65HbWOkxnprwqor3QoNdKpAM42L.jpeg">404not found|690x387</a></p>

**Comment by lee-dohm** [comment]
<aside class="quote no-group" data-username="captiongram" data-post="7" data-topic="115509">
<div class="title">
<div class="quote-controls"></div>
 captiongram:</div>
<blockquote>
<p>in that i have put 3 options “whatsapp”,“Facebook”,“Instagram” they are not working can u please help!!</p>
</blockquote>
</aside>
<p>The filenames that you use in the <code>href</code> attribute are <strong>case-significant</strong>. What this means is that <code>Whatsapp.html</code> is considered to be a different file from <code>whatsapp.html</code> because the <code>W</code> in the filename is a different case. So you need to change the text you’re using in your <code>href</code> attributes in <code>services.html</code> to match the actual filenames you have in your repository.</p>
<p>I created a <a href="https://github.com/captiongram/captiongram.github.io/pull/1">pull request</a> to show you what I mean. I hope that helps!</p>

**Comment by sanyam-2407** [comment]
<p>I am facing the same issue.Please help.<br>
i have a repository <a href="https://github.com/sanyam-2407/sanyamm" rel="nofollow noopener">https://github.com/sanyam-2407/sanyamm</a>] but when ever i open it ,they show 404 error not found can you please help me to publish my website its not publishing i waited for more than an hour.<br>
website link:<a href="https://sanyam-2407.github.io/sanyamm/" rel="nofollow noopener">https://sanyam-2407.github.io/sanyamm/</a><br>
</p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/3/3da355b3c2980c7f58e5264e4e2cfb48d066e6a0.jpeg" data-download-href="/uploads/short-url/8Nh4gBODAVPd5cHG4rlm9nzS1UI.jpeg?dl=1" title="git" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/65824616/181068900-b2843582-f0fc-4e6d-9287-99fcca285918.jpeg" alt="git" data-base62-sha1="8Nh4gBODAVPd5cHG4rlm9nzS1UI" width="690" height="388" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/3/3da355b3c2980c7f58e5264e4e2cfb48d066e6a0_2_690x388.jpeg, https://aws1.discourse-cdn.com/github/optimized/2X/3/3da355b3c2980c7f58e5264e4e2cfb48d066e6a0_2_1035x582.jpeg 1.5x, https://aws1.discourse-cdn.com/github/optimized/2X/3/3da355b3c2980c7f58e5264e4e2cfb48d066e6a0_2_1380x776.jpeg 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/3/3da355b3c2980c7f58e5264e4e2cfb48d066e6a0_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">git</span><span class="informations">1917×1079 75.8 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>

**Comment by ssc-gov** [comment]
<p>thank you for the help</p>

**Comment by Tavioking7862** [comment]
<p>It also helps me to do better with <a href="https://www.besttenrackets.com" rel="nofollow noopener">top ten racquets in tennis</a> but my error is not resolved totally …at somewhere it is open correctly and somewhere is not working…</p>
